lib/MPIPreferences/src/MPIPreferences.jl

```diff
@@ -157,96 +157,105 @@ function identify_abi(libmpi)
+        elseif occursin("CRAY MPICH", version_string)
+            impl = "CrayMPICH"
+            # "MPI VERSION    : CRAY MPICH version 7.7.10 (ANL base 3.2)\n"
+            if (m = match(r"CRAY MPICH version (\d+.\d+.\d+)", version_string)) !== nothing
+                version = VersionNumber(m.captures[1])
+            end
+        elseif startswith(version_string, "FUJITSU MPI")
+            impl = "FujitsuMPI"
+            # "FUJITSU MPI Library 4.0.0 (4.0.1fj4.0.0)\0"
+            if (m = match(r"^FUJITSU MPI Library (\d+.\d+.\d+)", version_string)) !== nothing
+                version = VersionNumber(m.captures[1])
+            end
+        elseif startswith(version_string, "MPIwrapper")
+            impl = "MPIwrapper"
+            # MPIwrapper 2.2.2
+            if (m = match(r"^MPIwrapper Version:\t(\d+.\d+.\d+\w*)", version_string)) !== nothing
                 version = VersionNumber(m.captures[1])
-            else
-                version = VersionNumber(m.captures[1]*"."*m.captures[2])
             end
         end
-
-    elseif startswith(version_string, "MVAPICH2")
-        impl = "MVAPICH"
-        # "MVAPICH2 Version      :\t%s\n")
-        if (m = match(r"^MVAPICH2? Version\s*:\t(\S*)\n", version_string)) !== nothing
-            version = VersionNumber(m.captures[1])
+        # 3) determine the abi from the implementation + version
+        if (impl == "MPICH" && version >= v"3.1" ||
+            impl == "IntelMPI" && version > v"2014" ||
+            impl == "MVAPICH" && version >= v"2" ||
+            impl == "CrayMPICH" && version >= v"7")
+            # https://www.mpich.org/abi/
+            abi = "MPICH"
+        elseif impl == "OpenMPI" || impl == "IBMSpectrumMPI" || impl == "FujitsuMPI"
+            abi = "OpenMPI"
+        elseif impl == "MicrosoftMPI"
+            abi = "MicrosoftMPI"
+        elseif impl == "MPIwrapper"
+            abi = "MPItrampoline"
+        else
+            abi = "unknown"
         end
 
-    elseif occursin("CRAY MPICH", version_string)
-        impl = "CrayMPICH"
-        # "MPI VERSION    : CRAY MPICH version 7.7.10 (ANL base 3.2)\n"
-        if (m = match(r"CRAY MPICH version (\d+.\d+.\d+)", version_string)) !== nothing
-            version = VersionNumber(m.captures[1])
-        end
-    elseif startswith(version_string, "FUJITSU MPI")
-        impl = "FujitsuMPI"
-        # "FUJITSU MPI Library 4.0.0 (4.0.1fj4.0.0)\0"
-        if (m = match(r"^FUJITSU MPI Library (\d+.\d+.\d+)", version_string)) !== nothing
-            version = VersionNumber(m.captures[1])
-        end
-    elseif startswith(version_string, "MPIwrapper")
-        impl = "MPIwrapper"
-        # MPIwrapper 2.2.2
-        if (m = match(r"^MPIwrapper Version:\t(\d+.\d+.\d+\w*)", version_string)) !== nothing
-            version = VersionNumber(m.captures[1])
+        @info "MPI implementation" libmpi version_string impl version abi
+
+        if impl == "OpenMPI" && v"4.1.2" ≤ version < v"5"
```

Comment:
I don't know the exact version numbers affected.

Also, I wanted to check the value of `ZES_ENABLE_SYSMAN` here, but it appears that hwloc/openmpi change it after dlopening the library, and trying to access it _before_ dlopening the library also causes a segmentation fault, too :confused:

Reply:
Before `dlopen` it should not cause a segmentation, after dlopen and before dlclose maybe?

Reply:
I'll try to build a reproducer later today.

Reply:
BTW, after dlopen and before dlcose it doesn't cause segmentation fault but `ZES_ENABLE_SYSMAN` is always set to 1, so not very useful to look at.  I tried to access it before dlopen with `get(ENV, "ZES_ENABLE_SYSMAN", "")` but I got the segmentation fault upon dlopen :confused:

Reply:
Yup....